test/extended/s2i_ruby_test.go

```diff
@@ -0,0 +1,81 @@
+// +build s2i
+
+package extended
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = Describe("OpenShift Ruby S2I image", func() {
+	defer GinkgoRecover()
+	var (
+		oc       = exutil.NewCLI("mysql-create", kubeConfigPath())
+		versions = map[string]string{
+			"20-centos7": "ruby 2.0.0",
+			//"20-rhel7":   "ruby 2.0.0",
+			//"22-centos7": "ruby 2.2.2",
+			//"22-rhel7":   "ruby 2.2.2",
+		}
+	)
+
+	for v, expected := range versions {
+
+		Describe(fmt.Sprintf("SCL usage of ruby-%s", v), func() {
+			It("should provide usage", func() {
+				oc.SetOutputDir(testContext.OutputDir)
+				name := exutil.MustGetImageName(oc.REST().ImageStreams("openshift"), "ruby", "latest")
```

Comment:
then we should just use DockerImage directly instead of IS
On Aug 7, 2015 5:04 PM, "Ben Parees" <notifications@github.com> wrote:

> In test/extended/s2i_ruby_test.go
> <https://github.com/openshift/origin/pull/4029#discussion_r36527626>:
>
> > +	var (
> > +		oc       = exutil.NewCLI("mysql-create", kubeConfigPath())
> > +		versions = map[string]string{
> > +			"20-centos7": "ruby 2.0.0",
> > +			//"20-rhel7":   "ruby 2.0.0",
> > +			//"22-centos7": "ruby 2.2.2",
> > +			//"22-rhel7":   "ruby 2.2.2",
> > +		}
> > +	)
> > +
> > +	for v, expected := range versions {
> > +
> > +		Describe(fmt.Sprintf("SCL usage of ruby-%s", v), func() {
> > +			It("should provide usage", func() {
> > +				oc.SetOutputDir(testContext.OutputDir)
> > +				name := exutil.MustGetImageName(oc.REST().ImageStreams("openshift"), "ruby", "latest")
>
> @mfojtik <https://github.com/mfojtik> I don't think that's going to do
> what you want, or not guaranteed anyway.
>
> --latest-images has no bearing on what builder image (eg ruby-20-centos7)
> the build flows use.
>
> The ImageStream is going to import from the registry when the imagestream
> is created (and is that the rhel7 or centos imagestream you're creating? or
> do you run the test twice, once for each imagestream?) which means it's
> going to set the imageid to whatever is in the registry at the time the
> imagestream was created.
>
> The build, when run, is going to resolve the ImageStreamTag to a pullspec
> which in this case might end in :latest in which case yes s2i will default
> to using the existing image if present, but at some point the
> ImageStreamTag might point to a pullspec which instead ends in a :imageid
> (v2 registry style) in which case the local image you just built will be
> totally ignored.
>
> and all that assumes we never set the BuildConfig forcePull setting to
> true.
>
> some further discussion w/ andy on irc:
>
> agoldste: if my imagestream points to a v2 registry
> (10:10:22 AM) bparees: agoldste: the dockerImageReference for a given
> ImageStreamTag will be of the form something/image@id right?
> (10:29:09 AM) bparees: agoldste: ok. so was my description of the behavior
> wrong?
> (10:29:33 AM) agoldste: bparees: it's correct, but we don't support
> importing from v2 registries yet
> (10:29:50 AM) agoldste: i.e. spec.dockerImageRepository =
> someV2registry/some/repo won't import any tags
> (10:30:09 AM) bparees: agoldste: ok.....what if the imagestream declares
> some tags in its spec? will we import those?
> (10:30:37 AM) agoldste: we need to rework pruning so the registry owns
> pruning its data (instead of the oadm prune images command making http
> prune requests to the registry), then we need to bump the
> docker/distribution code, then we need to modify the image import
> controller to support v2 imports
> (10:30:53 AM) agoldste: we can't import any data from a v2 registry right
> now
> (10:31:06 AM) agoldste: other than the direct integration w/have with our
> internal registry
>
> So basically once we move to v2 importing this logic will break because
> it'll try to run the imageid that exists in the registry, not the one you
> just built locally.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4029/files#r36527626>.
>